tests/unit/s2n_x509_validator_test.c

```diff
@@ -1803,17 +1803,53 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_pkey_zero_init(&public_key_out));
         s2n_pkey_type pkey_type = S2N_PKEY_TYPE_UNKNOWN;
 
-        /* Expect to return S2N_CERT_ERR_UNTRUSTED */
-        EXPECT_ERROR_WITH_ERRNO(s2n_x509_validator_validate_cert_chain(&validator, connection, chain_data, chain_len,
-                                        &pkey_type, &public_key_out),
-                S2N_ERR_CERT_UNTRUSTED);
+        EXPECT_ERROR_WITH_ERRNO(
+                s2n_x509_validator_validate_cert_chain(&validator,
+                        connection, chain_data, chain_len, &pkey_type, &public_key_out),
+                S2N_ERR_DECODE_CERTIFICATE);
 
         s2n_stuffer_free(&chain_stuffer);
         s2n_connection_free(connection);
         s2n_pkey_free(&public_key_out);
         s2n_x509_validator_wipe(&validator);
     };
 
+    /* Test unknown curve in cert validator for negative case */
+    if (s2n_libcrypto_is_awslc()) {
```

Comment:
> I could also test for success on openssl-1.1.1, but unless I'm forgetting some trick, specifically checking for openssl-1.1.1 is pretty tricky in our code

Would it help to limit the positive case to OpensSSL 3? Then there shouldn't be confusion between the libcrypto being OpenSSL and being a fork.

Reply:
Maybe, but I'm also a little worried about the curve being disabled / unavailable in some environments-- I saw some discussion around RHEL disabling some curves at build time, for example. I could maybe pick a different curve that's more popular but also not supported, but honestly I'm not sure I want us to depend on support for any curve we don't actually need.

Like I think at a minimum I'd want to check if the curve exists independent of this test. And that doesn't guarantee us any positive case, since it wouldn't be an error not to have the curve.

Reply:
Right ok, that makes sense.